### ISSUE TITLE
Compute Shader Skinning

### ISSUE THREAD

**Issue by DGriffin91** [bevyengine/bevy#7502]
## What problem does this solve or what need does it fill?

Currently Skinning is recomputed in many passes (prepass, main pass, shadow passes, etc...).

## What solution would you like?

Utilize compute shader skinning where skinning is done beforehand in a compute shader which writes out skinned vertices to a buffer. There are a few advantages to doing this:

- Skinning is only computed once for all passes
- Results in fewer shader permutations since vertex shaders are not specialized on skinning.
- The previous skinned vertex buffers can be kept around for computing the velocity pass (used in TAA, motion blur, etc...)

The use of compute shaders means that WebGL would require a separate skinning implementation.

## Additional context

Article about wicked engine's implementation: https://wickedengine.net/2017/09/09/skinning-in-compute-shader/
Doom Eternal uses this: https://simoncoenen.com/blog/programming/graphics/DoomEternalStudy

**Comment by nicopap** [comment]
Should include computation of morph targets in the compute pass as well.

**Comment by pcwalton** [comment]
Unity does this too.

**Comment by HydrogenC** [comment]
Is this featute still active and planned for implementation? Doing skin transfroms in every pass is indeed inefficient.

**Comment by DGriffin91** [comment]
I'm not aware of anyone working on it. But I'd guess it'll also be desirable for RT (maybe with some kind of BLAS cache?)

**Comment by HydrogenC** [comment]
> I'm not aware of anyone working on it. But I'd guess it'll also be desirable for RT (maybe with some kind of BLAS cache?)

Yes, to bring RT to skinned meshes would require this to be implemented.

**Comment by djeedai** [comment]
This would also be useful for random point sampling over the animated mesh, notably for VFX (Hanabi). Currently to implement that feature Hanabi would have to implement its own compute based skinning in parallel of Bevy's builtin one.
